server/src/lib/basket.ts

```diff
@@ -11,80 +11,154 @@ export const API_URL =
     ? 'https://basket.mozilla.org'
     : 'https://basket-dev.allizom.org';
 
+const basketConfig = {
+  uri: API_URL + '/news/common-voice-goals/',
+  method: 'POST',
+  headers: {
+    'x-api-key': BASKET_API_KEY,
+  },
+};
+
 function toISO(date: string) {
   return date ? new Date(date).toISOString().slice(0, -5) + 'Z' : null;
 }
 
-export async function sync(client_id: string) {
-  await computeGoals(client_id);
-
+async function getCurrentStatus(client_id: string) {
   const [[row]] = await db.query(
+    `SELECT u.*,
+      NOW() AS new_last_active,
+      HOUR(TIMEDIFF(NOW(), u.last_active)) AS hours_elapsed
+    FROM user_client_newsletter_prefs AS u
+    WHERE client_id = ?`,
+    [client_id]
+  );
+
+  return row;
+}
+
+function updateLastActive(currentUserStats: any) {
+  db.query(
+    `UPDATE user_client_newsletter_prefs
+      SET last_active = ?
+    WHERE client_id = ?`,
+    [currentUserStats.new_last_active, currentUserStats.client_id]
+  );
+
+  const data = {
+    email: currentUserStats.email,
+    last_active_date: toISO(currentUserStats.new_last_active),
+  };
+
+  sendRequest({
+    ...basketConfig,
+    form: data,
+  }).catch(err => console.error(err.message));
+
+  console.log('basket', JSON.stringify(data, null, 2));
+  return;
+}
+
+async function updateFullBasket(currentUserStats: any) {
+  const [[computed]] = await db.query(
     `
       SELECT
         email,
         basket_token,
         LEAST(
-          (SELECT MIN(clips.created_at) FROM clips WHERE client_id = user_clients.client_id),
-          (SELECT MIN(votes.created_at) FROM votes WHERE client_id = user_clients.client_id)
+          IFNULL((SELECT MIN(clips.created_at) FROM clips WHERE client_id = newsletter.client_id), NOW()),
+          IFNULL((SELECT MIN(votes.created_at) FROM votes WHERE client_id = newsletter.client_id), NOW())
```

Comment:
It turns out `LEAST()` will return NULL if one of the values is NULL, which means that if a user has voted but not recorded or vice versa it'll look like their first_contribution hasn't happened yet.

Reply:
Nice catch!